### PR TITLE
fix(testing): fix race condition in tests for parallel commands runner

### PR DESCRIPTION
**Pull request by rarmatei**
## Current Behavior (This is the behavior we have today, before the PR is merged)

Test would occasionally fail.

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

Test should consistently pass now.

## Issue

The previous setup was as follows:
- **command_1** was sleeping for 0.2 seconds before running
- **command_2** was sleeping for 0.1 seconds before running
The idea was that, if truly running in parallel, the second commands would finish before the first, because it was sleeping for less.

But as @jaysoo pointed out, if they're running in parallel we can't guarantee order, even with `sleep`.

## Fix (checked-in / part of this PR)

Instead of checking the order in which the commands were executed we now:
- invoke the commands runner synchronously
- immediately checked if there were 2 calls to spawn the processes
- wait for the promise/tasks to finish and check whether they both appended to the file (we do not check for order, just presence)

To prove above strategy works, it's also been added to the `serial` runner tests, and it indeed just prints 0 (as the initial promise it creates is a null one, only once that resolves it schedules the first task in the queue)

### CODE REVIEW

File: packages/workspace/src/builders/run-commands/run-commands.impl.spec.ts [hunk 1]
```diff
@@ -82,24 +86,29 @@ describe('Command Runner Builder', () => {
...
             },
             {
-              command: `sleep 0.1 && echo 2 >> ${f}`
+              command: `echo 2 >> ${f}`
             }
           ],
           parallel: true
         }
       );
+      const processesCreated = exec.calls.count();
```
Comment (FrozenPandaz) [inline packages/workspace/src/builders/run-commands/run-commands.impl.spec.ts]:
```suggestion
      await Promise.resolve();
      const processesCreated = exec.calls.count();
```

**Review by FrozenPandaz — APPROVED** [review]

File: packages/workspace/src/builders/run-commands/run-commands.impl.spec.ts (hunk 1 above)
Comment (jdpearce) [reply packages/workspace/src/builders/run-commands/run-commands.impl.spec.ts]: Is this comment in the right place? :)

Comment (rarmatei) [reply packages/workspace/src/builders/run-commands/run-commands.impl.spec.ts]: I added your suggestion to the serial runner tests 👍

**Review by FrozenPandaz — APPROVED** [review]
